File: TicTacToe/TicTacToePreferences.m
//
//  NSObject+TicTacToePreferences.m
//  TicTacToe
//
//  Created by Hélio Mesquita on 22/01/20.
//  Copyright © 2020 Hélio Mesquita. All rights reserved.
//

#import "TicTacToePreferences.h"

@implementation TicTacToePreferences

TicTacToeContainer * container;

+ (TicTacToePreferences *)shared {
    static TicTacToePreferences *shared = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        shared = [[self alloc] init];
    });
    return shared;
}

- (id)init {
    if (self = [super init]) {
        self.isUsingScenePattern = false;
        self.container = [[TicTacToeContainer new]init];
    }
    return self;
}

- (void)show {
    [self.container show];
}

- (void)remove {
    [self.container remove];
}

@end

File: TicTacToeExample/TicTacToePreferences.swift
//
//  JustPopupPreferences.swift
//  JustPopup
//
//  Created by Mefodiy Akatov on 24.07.2019.
//  Copyright © 2019 Eubicor. All rights reserved.
//

import Foundation

public class TicTacToePreferences {

    var isUsingScenePattern: Bool = false
    let container: TicTacToeContainer
    
    public static let shared = TicTacToePreferences()
    
    private init() {
        self.container = TicTacToeContainer()
    }

    func show() {
        container.show()
    }

    func remove() {
        container.remove()
    }

}
